libraries/AP_Param/AP_Param.cpp

```diff
@@ -2037,6 +2037,18 @@ void AP_Param::convert_old_parameters_scaled(const struct ConversionInfo *conver
     flush();
 }
 
+void AP_Param::assert_not_in_storage(const ConversionInfo *table, uint8_t table_size)
+{
+    for (uint8_t i=0; i<table_size; i++) {
+        const auto &info = table[i];
+        uint8_t old_value[type_size(info.type)];
+        AP_Param *ap = (AP_Param *)&old_value[0];
+        if (find_old_parameter(&info, ap)) {
+            AP_BoardConfig::config_error("Stale parameters in storage; wipe storage and reload");
```

Comment:
This is just unfriendly. I have lots of vehicles that have been upgraded from old versions of copter, forcing people to wipe is forcing people to potentially make a mistake for our own convenience. Much better to automatically remove the stale parameters if possible.